bound/typescript/src/http/exchanges/update-exchange.ts

```diff
@@ -0,0 +1,59 @@
+import { tbdexError } from "../../errors";
+import { Cancel } from "../../messages/cancel";
+import { Order } from "../../messages/order";
+import wasm from "../../wasm";
+
+export type WalletUpdateMessage = Order | Cancel;
+
+export class UpdateExchangeRequestBody {
+  readonly message: WalletUpdateMessage;
+
+  constructor(message: WalletUpdateMessage) {
+    this.message = message;
+  }
+
+  static fromWASM = (
+    wasmUpdateExchangeRequestBody: wasm.WasmUpdateExchangeRequestBody
+  ): UpdateExchangeRequestBody => {
+    try {
+      const kind = wasmUpdateExchangeRequestBody.data.kind;
+      const json = wasmUpdateExchangeRequestBody.data.json;
+
+      let message: WalletUpdateMessage;
+
+      if (kind === "order") message = Order.fromJSONString(json);
+      else if (kind === "cancel") message = Cancel.fromJSONString(json);
```

Comment:
is this the right spot to do logic?

shouldn't all logic basically be in rust?

Reply:
yeah good question, the TL;DR is the wasm FFI needs concrete types, and the `UpdateExchangeRequestBody` has a `message` property that can either be an `Order` or a `Cancel`, so JSON serialize it over the FFI. so we have to embed logic on the ts side to deserialize it back into the concrete type, such that then the DX has a true `WalletUpdateMessage` type, so they can do conditionals like `if (message instanceof Order) { ... } else if (message instanceof Cancel) { ... }`. this is one of those things I'd like to spend time optimizing if we ever have the time.